data-analytics/iot-nirvana/app-engine/src/main/java/com/google/cloud/demo/iot/nirvana/shared/Utils.java

```diff
@@ -0,0 +1,373 @@
+
+/** Class that describes functions used throughout the application */
+public class Utils {
+
+  private static final Logger LOG = Logger.getLogger(Utils.class.getName());
+
+  /**
+   * Convert an object to JSON
+   *
+   * @param obj the object to be encoded
+   * @return encoded object
+   */
+  public static String toJSON(Object obj) {
+    Gson gson = new Gson();
+    return gson.toJson(obj);
+  }
+
+  /**
+   * Convert a JSON string to an Object
+   *
+   * @param json the json to be converted
+   * @param destinationClass the object to be generated
+   * @return encoded object
+   */
+  public static <T> T fromJSON(String json, Class<T> destinationClass) {
+    Gson gson = new Gson();
+    return gson.fromJson(json, destinationClass);
+  }
+
+  /**
+   * Check if the app is running in production or in local dev
+   *
+   * @return true if it is in production, false otherwise
+   */
+  public static boolean isGaeProduction() {
+    return SystemProperty.environment.value() == SystemProperty.Environment.Value.Production;
+  }
+
+  /**
+   * Check if the running environment is the dev
+   *
+   * @return true if it is the dev, false otherwise
+   */
+  public static boolean isGaeDevEnvironment() {
+    return !isGaeProduction();
+  }
+
+  /**
+   * Get input parameters if they are stored in JSON format
+   *
+   * @param req
+   * @return
+   * @throws IOException
+   */
+  public static Map<String, String> getRequestParameters(HttpServletRequest req)
+      throws IOException {
+    // read input data
+    InputStream is = req.getInputStream();
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    byte[] buf = new byte[1024];
+    int r = 0;
+    while (r >= 0) {
+      r = is.read(buf);
+      if (r >= 0) {
+        os.write(buf, 0, r);
+      }
+    }
+
+    String inputParams = new String(os.toByteArray(), "UTF-8");
+    Gson gson = new Gson();
+
+    Type type = new TypeToken<Map<String, String>>() {}.getType();
+    Map<String, String> map = gson.fromJson(inputParams, type);
+
+    return map;
+  }
+
+  /**
+   * Get execution task of a task
+   *
+   * @param req the request of the task
+   * @return the execution time of a task if exist
+   */
+  public static int getExecutionCount(HttpServletRequest req) {
+    String header = req.getHeader("X-AppEngine-TaskExecutionCount");
+    if (header != null && !header.isEmpty()) {
+      return Integer.parseInt(header);
+    }
+    return 0;
+  }
+
+  /**
+   * Get execution task name
+   *
+   * @param req the request of the task
+   * @return the execution time of a task if exist
+   */
+  public static String getExecutionName(HttpServletRequest req) {
+    String header = req.getHeader("X-AppEngine-TaskName");
+    if (header == null) {
+      return null;
+    } else {
+      return header;
+    }
+  }
+
+  /**
+   * Convert a String array in a single string where all tokens are separated by a comma
+   *
+   * @param input the String array that you want to split
```

Comment:
"want to split"   --> want to join

Reply:
Entire fromArrayToString method removed, it was not used. Commit: https://github.com/GoogleCloudPlatform/professional-services/pull/61/commits/cf27a88180b74f1da3ee8eb53970fb5c4966bc88